File: easy/1189-maximum-number-of-balloons/solution.java
/**
 * 1189. Maximum Number of Balloons
 * Time: O(n)
 * Space: O(1)
 */
class Solution {
    public int maxNumberOfBalloons(String text) {
        int[] count = new int[26];
        for (int i = 0; i < text.length(); i++) {
            count[text.charAt(i) - 'a']++;
        }

        int answer = Math.min(count['b' - 'a'], count['a' - 'a']);
        answer = Math.min(answer, count['l' - 'a'] / 2);
        answer = Math.min(answer, count['o' - 'a'] / 2);
        answer = Math.min(answer, count['n' - 'a']);

        return answer;
    }
}

File: easy/1189-maximum-number-of-balloons/solution.py
from collections import Counter


class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        """
        1189. Maximum Number of Balloons
        Time: O(n)
        Space: O(1)
        """
        counts = Counter(text)
        return min(
            counts["b"],
            counts["a"],
            counts["l"] // 2,
            counts["o"] // 2,
            counts["n"],
        )
